### ISSUE TITLE
systemd-networkd: ipv4acd_on_packet() acd->state became IPV4ACD_STATE_STARTED and aborted on Hyper-V

### ISSUE THREAD

**Issue by yumetodo** [systemd/systemd#34489]
### systemd version the issue has been seen with

256.6

### Used distribution

Arch Linux

### Linux kernel version used

6.10.10-arch1-1

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Unexpected behaviour you saw

Frequently, systemd-networkd creates core dump and aborts. reboot doesn't resolve this issue.

Crashing interval is about to 5-10 min. 

### Additional program output to the terminal or log subsystem illustrating the issue

dump file, `coredumpctl dump` output, `jounalctl -x` log, `coredumpctl list` result, and Hyper-v's config is attached at gist shown below:

https://gist.github.com/yumetodo/fe4825ab623563d5b64e4710141eb66d

---

```
(gdb) backtrace
#0  0x000072a29baa53f4 in ?? () from /usr/lib/libc.so.6
#1  0x000072a29ba4c120 in raise () from /usr/lib/libc.so.6
#2  0x000072a29ba334c3 in abort () from /usr/lib/libc.so.6
#3  0x000072a29bc78416 in log_assert_failed_unreachable (file=file@entry=0x5b98333a81f3 "src/libsystemd-network/sd-ipv4acd.c", line=line@entry=407,
    func=func@entry=0x5b98333d2600 <__func__.2.lto_priv.100> "ipv4acd_on_packet") at ../systemd/src/basic/log.c:1004
#4  0x00005b983337b98e in ipv4acd_on_packet (s=<optimized out>, fd=<optimized out>, revents=<optimized out>, userdata=0x5b9854a693f0) at ../systemd/src/libsystemd-network/sd-ipv4acd.c:407
#5  0x000072a29bf0c2e2 in source_dispatch (s=0x5b9854a5f380) at ../systemd/src/libsystemd/sd-event/sd-event.c:4223
#6  0x000072a29bf0fa4e in sd_event_dispatch (e=<optimized out>, e@entry=0x5b9854a34ae0) at ../systemd/src/libsystemd/sd-event/sd-event.c:4844
#7  0x000072a29bf10121 in sd_event_run (e=<optimized out>, e@entry=0x5b9854a34ae0, timeout=timeout@entry=18446744073709551615) at ../systemd/src/libsystemd/sd-event/sd-event.c:4905
#8  0x000072a29bf10308 in sd_event_loop (e=<optimized out>) at ../systemd/src/libsystemd/sd-event/sd-event.c:4927
#9  0x00005b98332b605d in run (argc=<optimized out>, argv=<optimized out>) at ../systemd/src/network/networkd.c:109
#10 0x00005b98332ae7f4 in main (argc=<optimized out>, argv=<optimized out>) at ../systemd/src/network/networkd.c:116
(gdb) frame 4
#4  0x00005b983337b98e in ipv4acd_on_packet (s=<optimized out>, fd=<optimized out>, revents=<optimized out>, userdata=0x5b9854a693f0) at ../systemd/src/libsystemd-network/sd-ipv4acd.c:407
407                     assert_not_reached();
(gdb) list
402                     if (ipv4acd_arp_conflict(acd, &packet, false))
403                             ipv4acd_on_conflict(acd);
404                     break;
405
406             default:
407                     assert_not_reached();
408             }
409
410             return 0;
411
(gdb) p acd
$1 = (sd_ipv4acd *) 0x5b9854a693f0
(gdb) p acd->state
$2 = IPV4ACD_STATE_STARTED
```

`ipv4acd_on_packet` has a switch-case statement. However, it is not implemented on `acd->state == IPV4ACD_STATE_STARTED`.

```
(gdb) frame 4
#4  0x00005b983337b98e in ipv4acd_on_packet (s=<optimized out>, fd=<optimized out>, revents=<optimized out>, userdata=0x5b9854a693f0) at ../systemd/src/libsystemd-network/sd-ipv4acd.c:407
407                     assert_not_reached();
(gdb) p *acd
$9 = {n_ref = 1, state = IPV4ACD_STATE_STARTED, ifindex = 3, fd = 23, ifname = 0x0, n_iteration = 0, n_conflict = 0, receive_message_event_source = 0x5b9854a5f380, timer_event_source = 0x5b9854a67bf0,
  defend_window = 0, address = {s_addr = 16973481}, mac_addr = {ether_addr_octet = "\000\025]\000\a\005"}, event = 0x5b9854a34ae0, event_priority = 0, callback = 0x5b983331fe50 <on_acd>,
  userdata = 0x5b9854a58190, check_mac_callback = 0x5b98333206d0 <ipv4acd_check_mac>, check_mac_userdata = 0x5b9854a34de0}
```

Now, focused on `mac_addr = {ether_addr_octet = "\000\025]\000\a\005"}`, this could be decoded to `0:15:5d:0:7:5`.

https://wandbox.org/permlink/PP4aZA22aS3nCTa9

According to `ip n`, this is not `eth0` nor `eth1` network(???).

```
╰─$ ip n
172.19.64.1 dev eth0 lladdr 00:15:5d:6a:8c:b0 STALE
fe80::83db:9f98:3100:73bd dev eth0 lladdr 00:15:5d:6a:8c:b0 REACHABLE
fe80::deec:f0e9:bf2e:8b85 dev eth1 lladdr 00:15:5d:00:07:04 STALE
```

Hyper-V provides 2 switches.

![hyper-v_info](https://github.com/user-attachments/assets/ea6a39ab-3879-4cd8-9330-5ace407b484c)

---

### Host information

Windows 11 Pro 23H2 22631.4169

**Comment by yumetodo** [comment]
note

I have detected 704 arp packet in 5 min on intra switch(eth1).


https://gist.github.com/yumetodo/fe4825ab623563d5b64e4710141eb66d#file-20240919_intra-pcapng

**Comment by yuwata** [comment]
Thank you for the report. Fix is waiting #34503. If you'd like, please test the patch.
